LinkedList/LinkedList/TSLList.h

```diff
@@ -0,0 +1,350 @@
+			Size++;
+		}
+	}
+
+	//-------------------------------------------------
+	void AddTail(const Type& InTail)
+	{
+
+		if (Head == nullptr)
+		{
+			AddHead(InTail);
+		}
+		else
+		{
+			Tail->Next = new FNode;
+			Tail->Next->Element = InTail;
+			Tail = Tail->Next;
+			Size++;
+		}
+	}
+	
+	//----------------------------------------------------
+	void Insert(const Type& NewItem, const int Index)
+	{
+		if (Index == 0)
+		{
+			AddHead(NewItem);
+		}
+		else if(Index == Size)
+		{
+			AddTail(NewItem);
+		}
+
+		else if (Index > 0 && Index < Size)
+		{
+			FNode* Temporal = Head;
+			FNode* Container = new FNode;
+
+			for (int i = 0; i < Index; i++)
+			{
+				Temporal = Temporal->Next;
+			}
+			Container->Element = Temporal->Element;
+
+			Temporal->Element = NewItem;
+			Container->Next = Temporal->Next;
+			Temporal->Next = Container;
+			Size++;
+		}
+	}
+	//---------------------------------------------------------
+	void RemoveHead()
+	{
+		if (Size != 0)
+		{
+			FNode* NewHead = Head->Next;
+			delete Head;
+			Head = NewHead;
+			--Size;
+		}
+
+		else if (Size == 1)
+		{
+			delete Head;
+			Head = nullptr;
+			Tail = nullptr;
+			--Size;
+		}
+	}
+
+	//---------------------------------------------------------
+	void Remove(const int Index)
+	{
+		FNode* Temporal = Head;
+
+		if (Index == 0)
+		{
+			RemoveHead();
+		}
+
+		else if (Index >= 0 && Index <= Size)
+		{
+			for (int i = 0; i < Index - 1; i++)
+			{
+				Temporal = Temporal->Next;
+			}
+
+			FNode* Current = Temporal->Next;
+			Temporal->Next = Current->Next;
+
+			if (Current == Tail)
+			{
+				Tail = Temporal;
+			}
+			delete Current;
+			
+			Size--;
+		}
+	}
+
+	//----------------------------------------------------------
+	void Clear()
+	{
+		FNode* NextItem;
+
+		for (int i = 0; i < Size; i++)
+		{
+			NextItem = Head->Next;
+			delete Head;
+			Head = NextItem;
+		}
+		Size = 0;
+		Head = nullptr;
+		Tail = nullptr;
+	}
+
+	//----------------------------------------ForEach
+	template<typename Pred>
+	void forEach(const Pred& Predicate)
+	{
+		for (Type& Temporal: *this)
+		{
+			Predicate(Temporal);
+		}
+	}
+
+	//----------------------------------------Lambdas
+	template< typename Pred>
+	Type* FindByPredicate(const Pred& Predictate)
+	{
+		for (Type& Temporal : *this)
+		{
+			if (Predictate(Temporal))
+			{
+				return &Temporal->Element;
+			}
+		}
+		return nullptr;
+	};
+
+	//-------------------------------------------------
+	template< typename Pred>
+	TSLList<Type> FilterByPredicate(const Pred& Predicate)
+	{
+		FNode* Temporal;
+		TSLList <Type> Filterlist;
+
+		for(Type& Temporal : *this)
+		{
+			if (Predictate(Temporal))
+			{
+				Filterlist.AddTail(Temporal);		
+			}
+		}
+
+		return Filterlist;
+	};
+
+	//-------------------------------------------------
+	template< typename Pred>
+	void RemoveAllByPredicate(const Pred& Predicate)
+	{
+		FNode* Temporal = Head;
+		for (int i = 0; i < Size; i++)
+		{
+			if (Predictate(Temporal))
```

Comment:
Careful, parameter's name is Predicate but you're calling a Predictate function here